### PR TITLE
Add pre-commit hook for Linting

### PR DESCRIPTION
**Pull request by phillipbarron**
## What does this change?

This PR adds a pre-commit hook which will auto format staged changes using ScalaFMT. It also adds a setup script whose role is sym-linking the script to the hooks dir.

## How to test

Check it out, run the script, make a change and commit. The linter should run against your Scala files. 

## Have we considered potential risks?

Yep

### CODE REVIEW

**Comment by twrichards** [comment]
Ahhh inspired by your positive experiences of https://github.com/guardian/pinboard 😜

File: script/hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+ROOT=$(git rev-parse --show-toplevel)
+SCALAFMT_CONFIG="${ROOT}/.scalafmt.conf"
+
+if [ ! -f "${SCALAFMT_CONFIG}" ]; then
+  # Exit if no scalafmt configuration is found.
+  exit 0
+else
+  # Try the CLI version for performance.
+  echo "Checking code format via scalafmt..."
```
Comment (jonathonherbert) [inline script/hooks/pre-commit:13]: Just 'cos this is making changes

```suggestion
  echo "Formatting code format via scalafmt..."
```

File: script/hooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  echo "The check can be skipped by using the --no-verify flag for git commit."
```
Comment (jonathonherbert) [inline script/hooks/pre-commit:14]: As above

```suggestion
  echo "This step can be skipped by using the --no-verify flag for git commit."
```

**Review by jonathonherbert — APPROVED** [review]
Tested locally, great! Worth running ./setup-hooks in ./setup, so everyone new to the project gets the hooks?

**Comment by phillipbarron** [comment]
> Tested locally, great! Worth running ./setup-hooks in ./setup, so everyone new to the project gets the hooks?

Cheers - that was my plan as I has a memory of there being a  setup stcript but I don't see one 🤷

**Pull request merged**